KitePrintSDK/src/main/java/ly/kite/analytics/MixpanelAgent.java

```diff
@@ -0,0 +1,232 @@
+import android.util.Log;
+
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLConnection;
+
+import ly.kite.print.BaseRequest;
+
+
+///// Class Declaration /////
+
+/*****************************************************
+ *
+ * This class uploads analytics data to the Mixpanel
+ * API.
+ *
+ *****************************************************/
+public class MixpanelAgent implements BaseRequest.BaseRequestListener
+  {
+  ////////// Static Constant(s) //////////
+
+  @SuppressWarnings( "unused" )
+  private static final String  LOG_TAG             = "MixpanelAgent";
+
+  private static final String  ENDPOINT_URL_STRING = "https://api.mixpanel.com/track/";
+  public  static final String  API_TOKEN           = "e08854f70bc6a97b9f14457cbbb29b24";
+
+
+  ////////// Static Variable(s) //////////
+
+  private static MixpanelAgent  sMixpanelAgent;
+
+
+  ////////// Member Variable(s) //////////
+
+  private Context  mContext;
+
+
+  ////////// Static Initialiser(s) //////////
+
+
+  ////////// Static Method(s) //////////
+
+  /*****************************************************
+   *
+   * Returns a singleton instance of this class.
+   *
+   *****************************************************/
+  public static MixpanelAgent getInstance( Context context )
+    {
+    if ( sMixpanelAgent == null )
+      {
+      sMixpanelAgent = new MixpanelAgent( context );
+      }
+
+    return ( sMixpanelAgent );
+    }
+
+
+  ////////// Constructor(s) //////////
+
+  private MixpanelAgent( Context context )
+    {
+    mContext = context;
+    }
+
+
+  ////////// BaseRequest.BaseRequestListener Method(s) //////////
+
+  /*****************************************************
+   *
+   * Called when a request succeeds.
+   *
+   *****************************************************/
+  @Override
+  public void onSuccess( int httpStatusCode, JSONObject json )
+    {
+    // Ignore
+    }
+
+
+  /*****************************************************
+   *
+   * Called when a request fails.
+   *
+   *****************************************************/
+  @Override
+  public void onError( Exception exception )
+    {
+    Log.e( LOG_TAG, "Mixpanel request failed", exception );
+    }
+
+
+  ////////// Method(s) //////////
+
+  /*****************************************************
+   *
+   * Tracks an event.
+   *
+   *****************************************************/
+  public void trackEvent( JSONObject eventJSONObject )
+    {
+    // The JSON needs to be encoded as Base64
+
+    byte[] jsonBytes = eventJSONObject.toString().getBytes();
+
+    //Log.d( TAG, "JSON request:\n" + eventJSONObject.toString() );
+
+    String base64EncodedJSON = Base64.encodeToString( jsonBytes, Base64.NO_WRAP | Base64.URL_SAFE );
+
+
+    // Perform the HTTP request
+
+    String requestURLString = ENDPOINT_URL_STRING + "?data=" + base64EncodedJSON;
```

Comment:
could you also add `ip=1` query parameter (details in https://mixpanel.com/help/reference/http)? Mixpanel will do some nice stuff around augmenting the event with some additional ip related properties.